File: tests/data/flows/PedirInformacionCursoBuilder.ts
import { AccionFinalPedirInformacionCurso, EscenarioPedirInformacionCurso } from "./EscenarioPedirInformacionCurso";


export class PedirInformacionCursoBuilder {
  private escenario: EscenarioPedirInformacionCurso = {
    nombre: 'pedirInformacionCurso',
    descripcion:
      'Selecciona filtros en cursos graph, despliega un grupo y solicita información de un curso.',
    paginaInicial: '/cursos_graph.html',
    burbujas: [1, 2, 1],
    grupo: 2,
    accionFinal: 'solicitarPrimerCurso',
  };

  static unEscenario(): PedirInformacionCursoBuilder {
    return new PedirInformacionCursoBuilder();
  }

  conNombre(nombre: string): this {
    this.escenario.nombre = nombre;
    return this;
  }

  conDescripcion(descripcion: string): this {
    this.escenario.descripcion = descripcion;
    return this;
  }

  enPaginaInicial(paginaInicial: string): this {
    this.escenario.paginaInicial = paginaInicial;
    return this;
  }

  seleccionandoBurbujas(...burbujas: number[]): this {
    this.escenario.burbujas = burbujas;
    return this;
  }

  desplegandoGrupo(grupo: number): this {
    this.escenario.grupo = grupo;
    return this;
  }

  conAccionFinal(accionFinal: AccionFinalPedirInformacionCurso): this {
    this.escenario.accionFinal = accionFinal;
    return this;
  }

  build(): EscenarioPedirInformacionCurso {
    return {
      ...this.escenario,
      burbujas: [...this.escenario.burbujas],
    };
  }
}

File: playwright-dotnet/PlaywrightDotnet.Tests/Data/Flows/PedirInformacionCursoBuilder.cs
using System.Collections.Generic;

namespace PlaywrightDotnet.Tests.Data.Flows
{
    public class PedirInformacionCursoBuilder
    {
        private EscenarioPedirInformacionCurso escenario = new EscenarioPedirInformacionCurso();

        public static PedirInformacionCursoBuilder UnEscenario() => new PedirInformacionCursoBuilder();

        public PedirInformacionCursoBuilder ConNombre(string nombre)
        {
            escenario.Nombre = nombre;
            return this;
        }

        public PedirInformacionCursoBuilder ConDescripcion(string descripcion)
        {
            escenario.Descripcion = descripcion;
            return this;
        }

        public PedirInformacionCursoBuilder EnPaginaInicial(string paginaInicial)
        {
            escenario.PaginaInicial = paginaInicial;
            return this;
        }

        public PedirInformacionCursoBuilder SeleccionandoBurbujas(params int[] burbujas)
        {
            escenario.Burbujas = new List<int>(burbujas);
            return this;
        }

        public PedirInformacionCursoBuilder ConTags(string[] tags)
        {
            escenario.Tags = tags;
            return this;
        }

        public PedirInformacionCursoBuilder DesplegandoGrupo(int grupo)
        {
            escenario.Grupo = grupo;
            return this;
        }

        public PedirInformacionCursoBuilder ConAccionFinal(string accionFinal)
        {
            escenario.AccionFinal = accionFinal;
            return this;
        }

        public EscenarioPedirInformacionCurso Build()
        {
            // Clone simple fields to return a fresh instance
            return new EscenarioPedirInformacionCurso
            {
                Nombre = escenario.Nombre,
                Descripcion = escenario.Descripcion,
                PaginaInicial = escenario.PaginaInicial,
                Burbujas = new List<int>(escenario.Burbujas),
                Tags = escenario.Tags,
                Grupo = escenario.Grupo,
                AccionFinal = escenario.AccionFinal,
            };
        }
    }
}
